applications/nrf5340_audio/src/audio_module/src/amod_api.h

```diff
@@ -0,0 +1,498 @@
+/*
+ * Copyright(c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#ifndef _AMOD_API_H_
+#define _AMOD_API_H_
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <zephyr/kernel.h>
+#include "data_fifo.h"
+#include "aobj_api.h"
+
+/**
+ * @brief Maximum size for module naming in characters.
+ *
+ */
+#define AMOD_NAME_SIZE (20)
+
+/**
+ * @brief Modules private handle.
+ */
+struct handle;
+
+/**
+ * @brief Private context for the module.
+ */
+struct amod_context;
+
+/**
+ * @brief Modules private configuration.
+ */
+struct amod_configuration;
+
+/**
+ * @brief Module type relative to the module.
+ */
+enum amod_type {
+	/* The module type is undefined */
+	AMOD_TYPE_UNDEFINED = 0,
+
+	/* This is an input only module */
+	AMOD_TYPE_INPUT,
+
+	/* This is an output only module */
+	AMOD_TYPE_OUTPUT,
+
+	/* This is an input/output processing module */
+	AMOD_TYPE_PROCESSOR,
+};
+
+/**
+ * @brief Module state.
+ */
+enum amod_state {
+	/* Module state undefined */
+	AMOD_STATE_UNDEFINED = 0,
+
+	/* Module is in the open state */
+	AMOD_STATE_OPEN,
```

Comment:
When would the module be open and not configured? Is this state required?

Reply:
Will need to discuss, as this open state is a fall-back state if there are failures.